akka-actor/src/main/scala/akka/event/EventStreamUnsubscriber.scala

```diff
@@ -81,9 +75,13 @@ private[akka] object EventStreamUnsubscriber {
 
   def start(system: ActorSystem, stream: EventStream) = {
     val debug = system.settings.config.getBoolean("akka.actor.debug.event-stream")
-    system
+    val unsubscriber = system
       .asInstanceOf[ExtendedActorSystem]
       .systemActorOf(props(stream, debug), "eventStreamUnsubscriber-" + unsubscribersCount.incrementAndGet())
+    if (debug)
+      stream.publish(Logging.Debug(simpleName(getClass), getClass, s"registering unsubscriber with $stream"))
+    stream.initUnsubscriber(unsubscriber)
+    unsubscriber
```

Comment:
This was to fix #18630 but I think this could also explain the original failure in #30675 which is that the "watching" debug log from the `EventStreamUnsubscriber` wasn't seen, which could be explained by a race where the test started subscribing before the `EventStreamUnsubscriber` had not yet started and registered itself to the event stream.